services/workflows-service/src/metrics/repository/metrics.repository.ts

```diff
@@ -260,34 +260,99 @@ export class MetricsRepository {
-        GROUP BY
-          "riskLevel";`;
+  async getReportStatusesCount(projectId: TProjectId) {
+    const reportStatusCounts = await this.prismaService.businessReport.groupBy({
+      by: ['status'],
+      _count: {
+        status: true,
+      },
+      where: {
+        projectId,
+      },
+    });
+
+    const mappedStatuses = reportStatusCounts.reduce((acc, { _count, status }) => {
+      acc[status as keyof typeof BusinessReportStatus] = _count.status;
+
+      return acc;
+    }, {} as Record<keyof typeof BusinessReportStatus, number>);
+
+    const statusOrder: Array<keyof typeof BusinessReportStatus> = [
+      'completed',
+      'in_progress',
+      'new',
+      'failed',
+    ];
+
+    const result = statusOrder.map(status => ({
+      status: status == 'new' ? 'in_queue' : status,
+      count: mappedStatuses[status] || 0,
+    }));
 
-    return {
-      low: Number(results.find(result => result.riskLevel === 'low')?.count ?? 0),
-      medium: Number(results.find(result => result.riskLevel === 'medium')?.count ?? 0),
-      high: Number(results.find(result => result.riskLevel === 'high')?.count ?? 0),
-      critical: Number(results.find(result => result.riskLevel === 'critical')?.count ?? 0),
-    };
+    return result;
+  }
+
+  async getReportMCCsCount(projectId: TProjectId) {
+    const mccCounts = await this.prismaService.$queryRaw<Array<{ mcc: string; count: bigint }>>`
+    WITH "flattenedFormattedMCC" AS (
+      SELECT
+        "projectId",
+        CASE
+          WHEN jsonb_typeof("report"->'data'->'lineOfBusiness'->'formattedMcc') = 'string'
+          THEN jsonb_build_array("report"->'data'->'lineOfBusiness'->>'formattedMcc')
+          WHEN jsonb_typeof("report"->'data'->'lineOfBusiness'->'formattedMcc') = 'array'
+          THEN "report"->'data'->'lineOfBusiness'->'formattedMcc'
+          ELSE NULL
+        END AS mcc_array
+      FROM
+        "BusinessReport"
+      WHERE
+        "status"::text = ${BusinessReportStatus.completed}
+        AND "projectId" = ${projectId}
+    ),
+     "unnested" AS (
+      SELECT
+        "projectId",
+        jsonb_array_elements_text(mcc_array) AS mcc
+      FROM
+        "flattenedFormattedMCC"
+      WHERE
+        mcc_array IS NOT NULL
+    )
+    SELECT
+      mcc,
+      COUNT(*) AS count
+    FROM
+      "unnested"
+    WHERE
+      mcc IS NOT NULL AND mcc != ''
+    GROUP BY
+      mcc
+    ORDER BY
+      count DESC,
+      mcc ASC;
+  `;
+
+    const mccHash = Object.fromEntries(mccCounts.map(({ mcc, count }) => [mcc, Number(count)]));
+    const total = Object.values(mccHash).reduce((sum, count) => sum + count, 0);
+
+    return Object.entries(mccHash)
+      .map(([mccName, count]) => {
+        const [mcc, mccDescription] = mccName.split(' - ');
+
+        if (!mcc || !mccDescription) {
+          return;
+        }
+
+        return {
+          mcc: parseInt(mcc),
+          count,
+          percentage: Number(((count / total) * 100).toFixed(2)),
```

Comment:
Isn't it already a number?
```suggestion
          percentage: ((count / total) * 100).toFixed(2),
```

Reply:
to fixed makes it string, theefore number parsing it required